lib/rspecq/formatters/junit_formatter.rb

```diff
@@ -0,0 +1,43 @@
+require "rspec_junit_formatter"
+
+module RSpecQ
+  module Formatters
+    # Junit output formatter that handles outputting of requeued examples,
+    # the multiple suites per rspecq run.
+    class JUnitFormatter < RSpecJUnitFormatter
+      def initialize(queue, job, max_requeues, job_index)
+        @queue = queue
+        @job = job
+        @max_requeues = max_requeues
+        @requeued_examples = []
+        path = "test_results/results-#{ENV['TEST_ENV_NUMBER']}-#{job_index}.xml"
```

Comment:
You might already be thinking of this since this isn't finished, but I'd lean towards setting this as a CLI flag even if we don't upstream it just so that it's in config rather than code. Maybe this could just share the same flag as the one you have, if it was changed to `--junit-output` and accepted a string. If you wanted to automatically add the job number, it could accept a format string like `test_results/results-${NODE_INDEX}-%s.xml"` so the env var would be interpolated by bash but then you could use `sprintf` to add the job number, or something like that.